tests/test_hexary_trie_walk.py

```diff
@@ -304,10 +330,7 @@ def test_trie_walk_root_change_with_traverse(
         except TraversedPartialPath as exc:
             # You might only get part-way down a path of nibbles if your fog is based on an old trie
             # Determine the new sub-segments that are accessible from this partial traversal
-            sub_segments = [
-                exc.nibbles_traversed + next_segment
-                for next_segment in exc.node.sub_segments
-            ]
+            sub_segments = exc.simulated_node.sub_segments
```

Comment:
This entire PR is essentially motivated by wanting to enable this change. Not just in a couple tests, but as the "recommended" approach for handling `TraversedPartialPath` while walking a trie. (ie~ the readme changes)

The old approach was a bit cumbersome, but it was also broken. We didn't even notice because the code path wasn't touched! So most of the testing changes are about increasing the chance of catching this bug and other similar bugs. (Which did expose other issues in PRs that are now merged)